Add a first backup description (#353)
owncloud/docs-ocis: modules/ROOT/pages/maintenance/backup.adoc

= Backup
:toc: right

:description: Making a backup of Infinite Scale is an important task that should be done on a regular basis.

== Introduction

{description} This ensures that in case of issues, data can be restored from a backup.

== General Considerations

> Infinite Scale can run in two different setups when it comes to storing data.

Author: ```suggestion
Infinite Scale can run in two different setups when it comes to storing data:
```

> * All data, means configuration, user data, and metadata is stored on POSIX compliant filesystems.

Reviewer: ```suggestion
* All data, meaning configuration, user data and metadata, is stored on POSIX-compliant filesystems.
```

> * User data is stored as blob on an S3-compliant storage while configuration and metadata are stored on POSIX.

Reviewer 2: ```suggestion
* User data is stored as blobs on an S3-compliant storage while configuration and metadata are stored on POSIX.
```

> * User data is stored as blob on an S3 compliant storage and configuration respectively metadata is stored on POSIX

Reviewer: ```suggestion
* User data is stored as blob on an S3-compliant storage while configuration and metadata are stored on POSIX.
```

> For details on supported filesystems see: xref:prerequisites/prerequisites.adoc#filesystems-and-shared-storage[Filesystems and Shared Storage]

Reviewer: ```suggestion
For details on supported filesystems, see: xref:prerequisites/prerequisites.adoc#filesystems-and-shared-storage[Filesystems and Shared Storage].
```

> Depending on how this data is stored, backup strategies and tasks will differ, see the sections below.

Reviewer: ```suggestion
Depending on how this data is stored, backup strategies and tasks will differ (see the sections below).
```

> For the time being, you have to fully stop the Infinite Scale instance to create a consistant state necessary for taking the backup. In the scenarious shown, this consistancy state is a prerequisite for all upcoming tasks and the way to it is not explicit mentioned anymore. When the backup has finished, the Infinite Scale instance can be started again.

Reviewer: ```suggestion
You have to fully stop the Infinite Scale instance to create a consistent state necessary for performing the backup. In the following scenarios, this state of consistency is a prerequisite for all upcoming tasks. To avoid redundant information, this explanation will not be repeated every time. When the backup is finished, Infinite Scale can be started again.
```

Reviewer: The commands to stop and start Infinite Scale should be added here.

> You have to fully stop the Infinite Scale instance to create a consistent state necessary for performing the backup. In the following scenarios, this state of consistency is a prerequisite for all upcoming tasks. To avoid redundant information, this explanation will not be repeated every time. When the backup is finished, Infinite Scale can be started again.

Reviewer: The commands to stop and start Infinite Scale should be added here.

Author: There is no general "command" doing so as it depends on the installation method used (binary, container, helm).

> You have to fully stop the Infinite Scale instance to create a consistent state necessary for performing the backup. In the following scenarios, this state of consistency is a prerequisite for all upcoming tasks. To avoid redundant information, this explanation will not be repeated every time. When the backup is finished, Infinite Scale can be started again.

Reviewer: ```suggestion
You have to fully stop the Infinite Scale instance to create a consistent state necessary for performing the backup. In the following scenarios, this state of consistency is a prerequisite for all upcoming tasks. To avoid redundancy this step will not be repeated every time. When the backup is finished, Infinite Scale can be started again.
```

> You have to fully stop the Infinite Scale instance to create a consistent state necessary for performing the backup. In the following scenarios, this state of consistency is a prerequisite for all upcoming tasks. To avoid redundant information, this explanation will not be repeated every time. When the backup is finished, Infinite Scale can be started again.

Reviewer: ```suggestion
You have to fully stop the Infinite Scale instance to create a consistent state necessary for performing the backup. In the following scenarios, this state of consistency is a prerequisite for all upcoming tasks. To avoid redundancy this step will not be repeated every time. When the backup is finished, Infinite Scale can be started again.
```

> Note that creating a backup may only take a view seconds if the underlying filesystem of your storage provides snapshot capabilities. After taking the necessary snapshots, the Infinite Scale instance can be started again. To prevent data loss in case of hardware failure, we recommend copying the snapshots to a secondary storage.

Reviewer 2: ```suggestion
Note that creating a backup may only take a few seconds if the underlying filesystem of your storage provides snapshot capabilities. After taking the necessary snapshots, the Infinite Scale instance can be started again. To prevent data loss in case of hardware failure, we recommend copying the snapshots to a secondary storage.
```

> Note that taking a backup can be a task taking only a view seconds if the underlaying filesystem respectively storage is able to provide snapshot capabilities. After taking the necessary snapshots, the Infinite Scale instance can be started again. Based on the snapshots, any physical copy of data to a secondary storage can run independently by a follow up task.

Reviewer: ```suggestion
Note that creating a backup may only take a view seconds if the underlying filesystem of your storage provides snapshot capabilities. After taking the necessary snapshots, the Infinite Scale instance can be started again. To prevent data loss in case of hardware failure, we recommend copying the snapshots to a secondary storage.
```

> See the xref:deployment/general/general-info.adoc#default-paths[Default Paths] how locations are, respectively can be defined.

Reviewer: ```suggestion
See the xref:deployment/general/general-info.adoc#default-paths[Default Paths] for more information on where data is saved and how different locations can be defined.
```

> Keep in mind, that you always have to backup all three data sets consistantly which are:

Reviewer: ```suggestion
Keep in mind that you always have to back up all three data sets consistently:
```

* config data
* metadata ^1^
* user data ^1^

> (1) ... For POSIX-only environments, the location for user data and metadata is currently the same. When using S3, the location differs.

Author: ```suggestion
(1) ... For POSIX-only setups, the location for user data and metadata is currently the same. When using S3, the location differs.
```

> (1) ... For POSIX only environments, the location for user data and metadata is currently the same. When using S3, the location differs.

Reviewer: ```suggestion
(1) ... For POSIX-only environments, the location for user data and metadata is currently the same. When using S3, the location differs.
```

> == Pure POSIX Environment

Author: ```suggestion
== Pure POSIX Setup
```

> When all data, means configuration, user data and metadata are stored on POSIX compliant filesystems, create a consistancy state and backup the data sets. This is most easiest if you have onyl one filesystem for all of them. If you have different filesystems for the config and user/metadata, you need to do this task for each of them.

Reviewer: ```suggestion
If all data - configuration, user data and metadata - are stored on POSIX-compliant filesystems, create a consistent state and back up the data sets. This is easiest if you have only one filesystem for all of them. If you have different filesystems for the config and user/metadata, you need to do this task for each of them.
```

== Distributed Setup

> If data is distributed with configuration and metadata stored on POSIX-compliant filesystems and user data stored as blob on S3, create a consistent state and back up:

Reviewer 2: ```suggestion
If data is distributed with configuration and metadata stored on POSIX-compliant filesystems and user data stored as blobs on S3, create a consistent state and back up:
```

> When data is distributed, means configuration and metadata is stored on POSIX compliant filesystems and user data is stored as blob on S3, create a consistancy state and backup:

Reviewer: ```suggestion
If data is distributed with configuration and metadata stored on POSIX-compliant filesystems and user data stored as blob on S3, create a consistent state and back up:
```

* the configuration and metadata,

> * the S3 bucked according the rules and possibilities of the S3 provider.

Reviewer: ```suggestion
* the S3 bucket according to the rules and possibilities of the S3 provider.
```
